api/pkg/block/service.go

```diff
@@ -43,6 +47,163 @@ func NewAPIService(c client.Client) *APIService {
 	}
 }
 
+func UpdateFilter(reqFilter map[string]string, filter map[string]string) error {
+	log.Debugf("Getting the filters updated for the request.")
+
+	for k, v := range filter {
+		log.Debugf("Filter Key = [%+v], Key = [%+v]\n", k, v)
+		reqFilter[k] = v
+	}
+	return nil
+}
+
+func WriteError(response *restful.Response, msg string, errCode int, err error) {
+	log.Errorf(msg, err)
+	error := response.WriteError(errCode, err)
+	if error != nil {
+		log.Errorf("Response write status and http error failed: %v\n", err)
+		return
+	}
+}
+
+func (s *APIService) checkBackendExists(ctx context.Context, request *restful.Request,
```

Comment:
Suggestion  : Can we have a Test Connection kind of a abstraction , that is implemented for each vendor .. as there may be vendor specific way of testing the connectivity

Reply:
Just to clarify this a check for Cloud Backend resource existence in Backend Service. No Vendor Connectivity Checks here. :)